tests/integration/external_provider/stepper.py

```diff
@@ -0,0 +1,50 @@
+"""Stepper function for SIR model integration tests."""
+
+from typing import Any
+
+import numpy as np
+from numpy.typing import NDArray
+
+from flepimop2.system import SystemABC
+
+
+def stepper(
+    time: np.float64,  # noqa: ARG001
+    state: NDArray[np.float64],
+    **kwargs: Any,
+) -> NDArray[np.float64]:
+    """
+    ODE for an SIR model.
+
+    Args:
+        time: Current time (not used in this model).
+        state: Current state array [S, I, R].
+        **kwargs: Additional parameters (beta, gamma).
+
+    Returns:
+        Next state array after one step.
+    """
+    beta: float = kwargs.get("beta", 0.3)
+    gamma: float = kwargs.get("gamma", 0.1)
+    y_s, y_i, _ = np.asarray(state, dtype=float)
+    infection = beta * y_s * y_i / np.sum(state)
+    recovery = gamma * y_i
+    return np.array([-infection, infection - recovery, recovery], dtype=float)
+
+
+class SirSystem(SystemABC):
+    """SIR model system."""
+
+    def __init__(self) -> None:
+        """Initialize the SIR system with the SIR stepper."""
+        super().__init__(f=stepper)
+
+
+def build() -> SirSystem:
```

Comment:
same note as for engine

Reply:
https://github.com/ACCIDDA/flepimop2/pull/43#discussion_r2515541791